api/src/test/java/com/epam/pipeline/manager/pipeline/PipelineConfigurationManagerTest.java

```diff
@@ -16,182 +16,193 @@
         registry.setOwner(TEST_USER);
-        dockerRegistryDao.createDockerRegistry(registry);
-
-        library = new ToolGroup();
-        library.setName(TOOL_GROUP_NAME);
-        library.setRegistryId(registry.getId());
-        library.setOwner(TEST_USER);
-        toolGroupDao.createToolGroup(library);
+        registry.setId(10L);
 
         tool = new Tool();
         tool.setImage(TEST_IMAGE);
         tool.setRam(TEST_RAM);
         tool.setCpu(TEST_CPU);
         tool.setOwner(TEST_USER);
-
+        tool.setRegistry(registry.getPath());
         tool.setRegistryId(registry.getId());
-        tool.setToolGroupId(library.getId());
-        toolDao.createTool(tool);
 
-        // Data storages of user 1
-        NFSDataStorage dataStorage = new NFSDataStorage(dataStorageDao.createDataStorageId(), "testNFS",
-                                                        "test/path1");
+        NFSDataStorage dataStorage = new NFSDataStorage(TEST_DS_NFS_ID_1, "testNFS", "test/path1");
         dataStorage.setMountOptions("testMountOptions1");
         dataStorage.setMountPoint("/some/other/path");
         dataStorage.setOwner(TEST_OWNER1);
-        dataStorageDao.createDataStorage(dataStorage);
         dataStorages.add(dataStorage);
 
-        S3bucketDataStorage bucketDataStorage = new S3bucketDataStorage(dataStorageDao.createDataStorageId(),
-                                                                        "testBucket", "test/path2");
+        S3bucketDataStorage bucketDataStorage = new S3bucketDataStorage(TEST_DS_S3_ID_1, "testBucket", "test/path2");
         bucketDataStorage.setOwner(TEST_OWNER1);
-        dataStorageDao.createDataStorage(bucketDataStorage);
         dataStorages.add(bucketDataStorage);
 
         // Data storages of user 2
-        dataStorage = new NFSDataStorage(dataStorageDao.createDataStorageId(), "testNFS2", "test/path3");
+        dataStorage = new NFSDataStorage(TEST_DS_NFS_ID_2, "testNFS2", "test/path3");
         dataStorage.setMountOptions("testMountOptions2");
         dataStorage.setOwner(TEST_OWNER2);
-        dataStorageDao.createDataStorage(dataStorage);
         dataStorages.add(dataStorage);
 
-        bucketDataStorage = new S3bucketDataStorage(dataStorageDao.createDataStorageId(),
-                                                                        "testBucket2", "test/path4");
+        bucketDataStorage = new S3bucketDataStorage(TEST_DS_S3_ID_2, "testBucket2", "test/path4");
         bucketDataStorage.setOwner(TEST_OWNER2);
-        dataStorageDao.createDataStorage(bucketDataStorage);
         dataStorages.add(bucketDataStorage);
 
-        dataStorages.forEach(ds -> aclTestDao.createAclForObject(ds));
-        aclTestDao.grantPermissions(dataStorage, TEST_OWNER1,
-                                    Collections.singletonList((AclPermission) AclPermission.READ));
+        //let's suppose, that we granted READ permission to both owners: TEST_OWNER1 and TEST_OWNER2
```

Comment:
But owners already have full permissions for their entities.

Reply:
Fixed